src/db_queries/expenses.ts

```diff
@@ -0,0 +1,120 @@
+                year: sourceExpense.year || targetExpense.year,
+                expense: sourceExpense.expense || targetExpense.expense,
+                amount: sourceExpense.amount || targetExpense.amount,
+                is_recurring: sourceExpense.is_recurring === undefined ? targetExpense.is_recurring : sourceExpense.is_recurring,
+        } as CenterExpenses
+    }
+
+    async getCenterExpenses(request: Request, response: Response): Promise<void> {
+        const centerId = request.params.center_id;
+
+        try {
+            const result = await this.getCenterExpenseQuery(centerId);
+
+            if(result && result.rowCount === 0) {
+                response.status(404).send("No expenses found");
+            }
+
+            response.status(200).json(result?.rows);
+        } catch (error) {
+            response.status(400).send(error);
+        }
+    }
+
+    async createCenterExpenses(request: Request, response: Response): Promise<void> {
+        const {center_id} = request.params;
+        const { month, year, expense, amount,is_recurring } = request.body as CenterExpenses;
+        try {
+            await dbClient.query(this.createExpenseQuery, [center_id, month, year, expense, amount,is_recurring]);
+            response.status(201).send("Expense is created.");
+        } catch (error) {
+            response.status(400).send(error);
+        }
+    }
+
+    async deleteCenterExpenses(request: Request, response: Response): Promise<void> {
+        const {center_id, expense_id} = request.params;
+        try {
+            await dbClient.query(this.deleteExpenseQuery, [center_id, expense_id]);
+            response.status(200).send("Expense is deleted.");
+        } catch (error) {
+            response.status(400).send(error);
+        }
+    }
+
+    async updateCenterExpenses(request: Request, response: Response): Promise<void> {
+        // Todo: Add validation
+        const {center_id, expense_id} = request.params;
+
+        const expenseResult = await this.getCenterExpenseQuery(center_id);
+
+        if(expenseResult.rowCount === 0) {
+            response.status(404).send("No expenses found");
+            return;
+        }
+
+        const { month, year, expense, amount, is_recurring } = this.mergeExpense(expenseResult.rows[0], request.body as CenterExpenses)
+
+        try {
+            await dbClient.query(this.updateExpenseQuery, [month, year, expense, amount,is_recurring, center_id, expense_id]);
+            response.status(200).send("Expense is updated.");
+        } catch (error) {
+            response.status(400).send(error);
+        }
+    }
+
+    async getCenterExpenseBalance(request: Request, response: Response): Promise<void> {
+        const {center_id} = request.params;
+        try{
+            dbClient.withClient(async (client) => {
+                try {
+                    const totalExpense = await client.query(this.totalExpenseQuery, [center_id]);
```

Comment:
Plural works better here too: `totalExpenses` instead of `totalExpense` and `totalExpensesPaid` instead of `totalExpensePaid`.